internal/logger/default_logger.go

```diff
@@ -44,43 +44,39 @@ const (
 // If loggerConfig.SetLogger() method is called, the LoggingLevel property will not be used.
 type DefaultLogger struct {
 	*log.Logger
-	Level logger.Level
+	Level int
 }
 
 // New returns a Default Logger with defaultLogLevel.
 func New() *DefaultLogger {
+	numericLevel, _ := logger.GetLogLevel(defaultLogLevel) // defaultLogLevel exists, no err
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
-		Level:  defaultLogLevel,
+		Level:  numericLevel,
 	}
 }
 
-func NewWithLevel(loggingLevel logger.Level) *DefaultLogger {
+func NewWithLevel(loggingLevel logger.Level) (*DefaultLogger, error) {
+	numericLevel, err := logger.GetLogLevel(loggingLevel)
+	if err != nil {
+		return nil, err
+	}
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
-		Level:  loggingLevel,
-	}
+		Level:  numericLevel,
+	}, nil
 }
 
-func (l *DefaultLogger) canLog(level logger.Level) bool {
+func (l *DefaultLogger) Log(level logger.Level, formatter func() string) {
 	numericLevel, err := logger.GetLogLevel(level)
```

Comment:
Let's rename `numericLevel` to something better. How about `wantedLevel`?

Reply:
Sure 👍